lib/server/devicestatus.js

```diff
@@ -69,6 +70,72 @@ function storage (collection, ctx) {
     return api( ).remove(filter, fn);
   }
 
+  function removeOld (date, fn) {
+    var filter;
+    var year;
+    var endMonth;
+    var month;
+    var monthStr;
+    var day;
+    var dayStr;
+    var retVal = null;
+
+    var firstErr = null,
+        numDeletes = 0,
+        totalDeletes = 0,
+        totalRecordsDeleted = 0;
+
+    var endDate = moment(date);
+    endMonth = endDate.month() + 1;
```

Comment:
I'm not seeing how this stops at one month ago.  The way I'm reading it, it deletes up to the current month and day?  What am I missing?

Reply:
@scottleibrand, the date argument to the function is the day to delete up to. `moment(date)` converts the date string to a moment object so if the user sent an HTTP DELETE request to `$NS_URL/api/v1/devicestatus_delete/2018-02-12`, `endDate` would be set to `02/12/2018`.